step_create_snapshot.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"github.com/mitchellh/multistep"
+	"github.com/hashicorp/packer/packer"
+	"context"
+	"github.com/vmware/govmomi/object"
+)
+
+type StepCreateSnapshot struct{
+	createSnapshot bool
+	vmName         string
+}
+
+func (s *StepCreateSnapshot) Run(state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+	vmSrc := state.Get("vmSrc").(*object.VirtualMachine)
+	ctx := state.Get("ctx").(context.Context)
+
+	if s.createSnapshot {
+		ui.Say("creating snapshot...")
+
+		_, err := vmSrc.CreateSnapshot(ctx, s.vmName, "", true, true)
```

Comment:
the second parameter is not a vm name, that's a snapshot name. we can hardcode it with something like `Created by Packer` string